### PR TITLE
Fix consistency in the output of scalar tensors

### PR DESCRIPTION
**Pull request by SkafteNicki**
## What does this PR do?

Fixes https://github.com/PyTorchLightning/metrics/issues/611
Trying to fix consistency in the output of metrics. If a metric returns a scalar tensor, then sometimes it is
```python
tensor([...])
```
and other times it is:
```python
tensor(...)
```
This PR fixes this by doing a squeeze after `compute` if the output has `numel=1`.

## Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: torchmetrics/aggregation.py [hunk 1]
```diff
@@ -398,7 +398,7 @@ class MeanMetric(BaseAggregator):
         >>> metric.update(1)
         >>> metric.update(torch.tensor([2, 3]))
         >>> metric.compute()
-        tensor([2.])
+        tensor(2.])
```
Comment (SkafteNicki) [inline torchmetrics/aggregation.py:401]:
```suggestion
        tensor(2.)
```

**Review by Borda — APPROVED** [review]

File: torchmetrics/utilities/data.py [hunk 2]
```diff
@@ -238,3 +238,11 @@ def get_group_indexes(indexes: Tensor) -> List[Tensor]:
...
 
     return [tensor(x, dtype=torch.long) for x in res.values()]
+
+
+def _squeeze_scalar_element_tensor(x: Tensor) -> Tensor:
+    return x.squeeze() if x.numel() == 1 else x
+
+
+def _squeeze_if_scalar(data: Any) -> Any:
+    return apply_to_collection(data, Tensor, _squeeze_scalar_element_tensor)
```
Comment (carmocca) [inline torchmetrics/utilities/data.py:248]: noob q: why don't we update as many of the following to use `torch.tensor(0)`

https://grep.app/search?q=default%3Dtorch.zeros%28&filter[repo][0]=PyTorchLightning/metrics

Comment (Borda) [reply torchmetrics/utilities/data.py:248]: @SkafteNicki ^^

**Review by awaelchli — APPROVED** [review]

File: torchmetrics/utilities/data.py (hunk 2 above)
Comment (SkafteNicki) [reply torchmetrics/utilities/data.py:248]: Updated now. Still believe that a global squeeze is good to have to secure that we not in the future mess up.

File: torchmetrics/image/lpip_similarity.py [hunk 3]
```diff
@@ -127,8 +127,8 @@ def __init__(
             raise ValueError(f"Argument `reduction` must be one of {valid_reduction}, but got {reduction}")
         self.reduction = reduction
 
-        self.add_state("sum_scores", torch.zeros(1), dist_reduce_fx="sum")
-        self.add_state("total", torch.zeros(1), dist_reduce_fx="sum")
+        self.add_state("sum_scores", torch.tensor(0), dist_reduce_fx="sum")
+        self.add_state("total", torch.tensor(0), dist_reduce_fx="sum")
```
Comment (SkafteNicki) [inline torchmetrics/image/lpip_similarity.py:131]:
```suggestion
        self.add_state("sum_scores", torch.tensor(0.0), dist_reduce_fx="sum")
        self.add_state("total", torch.tensor(0.0), dist_reduce_fx="sum")
```

File: torchmetrics/regression/pearson.py [hunk 4]
```diff
@@ -108,12 +108,12 @@ def __init__(
...
-        self.add_state("var_x", default=torch.zeros(1), dist_reduce_fx=None)
-        self.add_state("var_y", default=torch.zeros(1), dist_reduce_fx=None)
-        self.add_state("corr_xy", default=torch.zeros(1), dist_reduce_fx=None)
-        self.add_state("n_total", default=torch.zeros(1), dist_reduce_fx=None)
+        self.add_state("mean_x", default=torch.tensor(0), dist_reduce_fx=None)
+        self.add_state("mean_y", default=torch.tensor(0), dist_reduce_fx=None)
+        self.add_state("var_x", default=torch.tensor(0), dist_reduce_fx=None)
+        self.add_state("var_y", default=torch.tensor(0), dist_reduce_fx=None)
+        self.add_state("corr_xy", default=torch.tensor(0), dist_reduce_fx=None)
+        self.add_state("n_total", default=torch.tensor(0), dist_reduce_fx=None)
```
Comment (SkafteNicki) [inline torchmetrics/regression/pearson.py:116]:
```suggestion
        self.add_state("mean_x", default=torch.tensor(0.0), dist_reduce_fx=None)
        self.add_state("mean_y", default=torch.tensor(0.0), dist_reduce_fx=None)
        self.add_state("var_x", default=torch.tensor(0.0), dist_reduce_fx=None)
        self.add_state("var_y", default=torch.tensor(0.0), dist_reduce_fx=None)
        self.add_state("corr_xy", default=torch.tensor(0.0), dist_reduce_fx=None)
        self.add_state("n_total", default=torch.tensor(0.0), dist_reduce_fx=None)
```

File: torchmetrics/classification/kl_divergence.py [hunk 5]
```diff
@@ -92,10 +92,10 @@ def __init__(
         self.reduction = reduction
 
         if self.reduction in ["mean", "sum"]:
-            self.add_state("measures", torch.zeros(1), dist_reduce_fx="sum")
+            self.add_state("measures", torch.tensor(0), dist_reduce_fx="sum")
```
Comment (SkafteNicki) [inline torchmetrics/classification/kl_divergence.py:95]:
```suggestion
            self.add_state("measures", torch.tensor(0.0), dist_reduce_fx="sum")
```

File: torchmetrics/aggregation.py [hunk 6]
```diff
@@ -410,9 +410,9 @@ def __init__(
         dist_sync_fn: Callable = None,
     ):
         super().__init__(
-            "sum", torch.zeros(1), nan_strategy, compute_on_step, dist_sync_on_step, process_group, dist_sync_fn
+            "sum", torch.tensor(0), nan_strategy, compute_on_step, dist_sync_on_step, process_group, dist_sync_fn
         )
-        self.add_state("weight", default=torch.zeros(1), dist_reduce_fx="sum")
+        self.add_state("weight", default=torch.tensor(0), dist_reduce_fx="sum")
```
Comment (SkafteNicki) [inline torchmetrics/aggregation.py:415]:
```suggestion
        self.add_state("weight", default=torch.tensor(0.0), dist_reduce_fx="sum")
```

File: torchmetrics/aggregation.py [hunk 7]
```diff
@@ -410,9 +410,9 @@ def __init__(
         dist_sync_fn: Callable = None,
     ):
         super().__init__(
-            "sum", torch.zeros(1), nan_strategy, compute_on_step, dist_sync_on_step, process_group, dist_sync_fn
+            "sum", torch.tensor(0), nan_strategy, compute_on_step, dist_sync_on_step, process_group, dist_sync_fn
```
Comment (SkafteNicki) [inline torchmetrics/aggregation.py:413]:
```suggestion
            "sum", torch.tensor(0.0), nan_strategy, compute_on_step, dist_sync_on_step, process_group, dist_sync_fn
```

File: torchmetrics/aggregation.py [hunk 8]
```diff
@@ -287,7 +287,7 @@ def __init__(
...
```
Comment (SkafteNicki) [inline torchmetrics/aggregation.py:290]:
```suggestion
            "sum", torch.tensor(0.0), nan_strategy, compute_on_step, dist_sync_on_step, process_group, dist_sync_fn
```
